engine/apps/public_api/tests/test_on_call_shifts.py

```diff
@@ -165,6 +165,52 @@ def test_get_override_on_call_shift(make_organization_and_user_with_token, make_
     assert response.data == result
 
 
+@pytest.mark.django_db
+def test_get_web_override_shift(
+    make_organization_and_user_with_token, make_user_for_organization, make_on_call_shift, make_schedule
+):
+    organization, user, token = make_organization_and_user_with_token()
+    other_user = make_user_for_organization(organization)
+    client = APIClient()
+
+    start_data = timezone.now().replace(microsecond=0)
+    schedule = make_schedule(organization, schedule_class=OnCallScheduleCalendar)
+    data = {
+        "name": "override shift",
+        "start": start_data,
+        "rotation_start": start_data,
+        "duration": timezone.timedelta(hours=9),
+        "schedule": schedule,
+    }
+    on_call_shift = make_on_call_shift(
+        organization=organization,
+        source=CustomOnCallShift.SOURCE_WEB,
+        shift_type=CustomOnCallShift.TYPE_OVERRIDE,
+        **data,
+    )
+    on_call_shift.add_rolling_users([[other_user], [user]])
+
+    url = reverse("api-public:on_call_shifts-detail", kwargs={"pk": on_call_shift.public_primary_key})
+
+    response = client.get(url, format="json", HTTP_AUTHORIZATION=f"{token}")
+
+    result = {
+        "id": on_call_shift.public_primary_key,
+        "team_id": None,
+        "schedule": schedule.public_primary_key,
+        "name": on_call_shift.name,
+        "type": "override",
+        "time_zone": None,
+        "start": on_call_shift.start.strftime("%Y-%m-%dT%H:%M:%S"),
+        "rotation_start": on_call_shift.start.strftime("%Y-%m-%dT%H:%M:%S"),
+        "duration": int(on_call_shift.duration.total_seconds()),
+        "users": list({user.public_primary_key, other_user.public_primary_key}),
```

Comment:
Note this may return users in any order, and then the test may [randomly fail](https://github.com/grafana/oncall/actions/runs/6878380832/job/18715666408?pr=3303). It will be nice to make the test stable.

Reply:
Have made it to use one user to avoid adding unnecessary sort function in the actual code.